setup.py

```diff
@@ -73,40 +63,13 @@ def run(self):
 if platform.machine() == 'ppc64le':
     extra_link_args += ['-pthread']
 
-build_ext = cmdclass.get('build_ext', build_ext)
-
-numba_be_user_options = [
-    ('werror', None, 'Build extensions with -Werror'),
-    ('wall', None, 'Build extensions with -Wall'),
-    ('noopt', None, 'Build extensions without optimization'),
-]
-
 
 class NumbaBuildExt(build_ext):
-
-    user_options = build_ext.user_options + numba_be_user_options
-    boolean_options = build_ext.boolean_options + ['werror', 'wall', 'noopt']
-
-    def initialize_options(self):
-        super().initialize_options()
-        self.werror = 0
-        self.wall = 0
-        self.noopt = 0
-
     def run(self):
-        extra_compile_args = []
-        if self.noopt:
-            if sys.platform == 'win32':
-                extra_compile_args.append('/Od')
-            else:
-                extra_compile_args.append('-O0')
-        if self.werror:
-            extra_compile_args.append('-Werror')
-        if self.wall:
-            extra_compile_args.append('-Wall')
+        optimization_level = "/Od" if sys.platform == 'win32' else "-O0"
+        extra_compile_args = ["-Werror", "-Wall", optimization_level]
```

Comment:
I think we don't have these on in all cases and instead guard them behind the `-wall` and `-werror` flags because some users might have a newer C compiler than we use / test with, which triggers unexpected warnings and causes the build to fail.